packages/core/src/config/parse.ts

```diff
@@ -1264,13 +1287,23 @@ permission to call the 'upgradeTo' function on each of them.
     console.error = () => {}
 
     // fetch the contract and validate
-    const upgradableContract = getOpenZeppelinUpgradableContract(
-      contractConfig.contract,
-      input,
-      output,
-      contractConfig.proxyType,
-      userContractConfig
-    )
+    // We use a try catch and then rethrow any errors because we temporarily disabled console.error
+    let upgradableContract: UpgradeableContract | undefined
+    try {
+      upgradableContract = getOpenZeppelinUpgradableContract(
+        contractConfig.contract,
+        input,
+        output,
+        // Since 'no-proxy' was introduced by us and not supported by OpenZeppelin, we need
+        // to pass in 'internal-default' instead so we can still run the validation on non-proxied contracts
+        contractConfig.kind !== 'no-proxy'
+          ? contractConfig.kind
+          : 'internal-default',
+        userContractConfig
+      )
+    } catch (e) {
+      throw e
+    }
```

Comment:
We can't skip it, we need it for the storage slot comparison.

Reply:
The try/catch here doesn't have anything to do with non-proxied contracts. I added it in because I realized that by overriding console.error, we're introducing the possibility that some unexpected error will be thrown and we miss it which could make debugging difficult.

As far as running the `getOpenZeppelinUpgradableContract`, IMO we should still run it for non-proxied contracts because there are checks that we do that are relevant to them. I.e checks for self-destruct and delegatecall. I think we should be consistent about the checks we're running for all contracts.

Reply:
I think it's worth spending some time thinking about how we can clean this up, but let's do it in a future PR. I made a ticket for it. For this PR, I think the only change to this code snippet is to skip calling `getOpenZeppelinUpgradableContract` for non-proxied contracts.